latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -159,4 +160,62 @@ public String render(String templateName) throws Exception
     {
         return render(getTemplate(templateName));
     }
+
+    private String getTemplateName(Block block)
+    {
+        // If there's a custom template defined in the Block parameter's, use it!
+        String templateName = block.getParameter("latex-template");
+        if (templateName == null) {
+            templateName = block.getClass().getSimpleName();
+        }
+        return templateName;
+    }
+
+    private Optional<String> renderUIXPs(String templateName, String suffix)
+    {
+        // The XDOM template case is particular, the after UIXP needs to be located before "\end{document}" which closes
+        // the latex document. To do so, the UIXP of XDOM are directly integrated in the template in Velocity.
+        Optional<String> result;
+        if (!templateName.equals("XDOM")) {
+            List<UIExtension> extensions =
+                this.uiExtensionSupplier.getExtensions(
+                    String.format("org.xwiki.contrib.latex.%s.%s", templateName, suffix));
+            if (extensions.isEmpty()) {
+                result = Optional.empty();
+            } else {
+                DefaultWikiPrinter printer = new DefaultWikiPrinter();
+                // Sort the extensions by their "order" parameter, the values with the highest order value first.
+                // Missing or invalid order values are considered to be 0.
+                extensions.sort((e1, e2) -> getOrderParameter(e2) - getOrderParameter(e1));
```

Comment:
I think the right way is to use https://github.com/xwiki/xwiki-platform/blob/457225b1c8e27c49e3b1f7d4fa2124eb867b2fda/xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/script/UIExtensionScriptService.java#L117

Since this is a SS and it's not in `UIExtensionManager` you won't be able to use it and you'll need to either 1) add the method to `UIExtensionSupplier` (which should maybe be renamed as a result) or 2) Introduce a `UIExtensionManager` component with a higher priority than `1000` and that adds the missing method, + of course add a jira to add a new api in the original `UIExtensionManager` component...

WDYT?

Reply:
FTR example usage: https://github.com/xwiki/xwiki-platform/blob/7cc61afd14e27b82a12d4388cb2e0e21e749fb57/xwiki-platform-core/xwiki-platform-alerts/xwiki-platform-alerts-ui/src/main/resources/XWiki/Alerts/Code/AlertsMenuUIX.xml#L120

Reply:
I'll rename, `UIExtensionSupplier` to `UIExtensionManager` and add a `List<UIExtension> get(String extensionPointId, Map<String, String> filters)` method.
I've created https://jira.xwiki.org/browse/XWIKI-19298 for XS.